.github/workflows/broken-linkcheck.yml

```
@@ -0,0 +1,30 @@
+---
+name: Broken linkcheck
+on:
+    workflow_dispatch: # Enable manual triggering
+    #schedule:
+    # Run monthly on the 1st day of the month
+        #- cron: '0 0 1 * *'
```

What @mroeschke advocates here is to also have a trigger for this PR to run this. You can do that by

```suggestion
        #- cron: '0 0 1 * *'
    pull_request:
```

With that, the check would be run on every PR. This is not what we want. Instead we should only run it if the PR touches a file that is directly linked to the test that we are running. My suggestion here is

```suggestion
        #- cron: '0 0 1 * *'
    pull_request:
      paths:
        - ".github/workflows/broken-linkcheck.yml"
        - "doc/make.py"  
```

This way we always run the check if someone touches the workflow file or modifies the make script and thus potentially touches the linkcheck functionality.

---

I have a doubt. Why we are using `"- doc/make.py" ` in the `paths:` ?

---

Because if someone changes and accidentally breaks the linkcheck command in `doc/make.py` the CI would not flag it in the PR. I'll leave it up to the maintainers to decide if they want this or not.

---

> Because if someone changes and accidentally breaks the linkcheck command in `doc/make.py` the CI would not flag it in the PR.

Didn't understood it properly. Could you explain this little more ?